docs/DeveloperGuide.md

```diff
@@ -1,29 +1,215 @@
+Other than that, the way the functions work is the same.
+
+Below is the explanation of the implementation of based on buy stock command, 
+which can be applied to the sell stock command as well.
+
+Given below is an example usage scenario and how buy stock command behaves at each step.
+
+![](./diagrams/BuyStockState0.png)
+
+**Step 1** : The user calls the buy stock command from the `Parser`. 
+Buy stock command is instantiated by `Parser`, which calls the `buyParse` method to get the attribute values 
+for `symbol` and `quantity` required to instantiate `buyCommand`. The `Controller` instantiates `Ui`, `PortfolioManager`
+and `StockPriceFetcher`, which gets the `price` of stock. 
+In the `Controller`, we check for the instance of `buyCommand` object and calls the `buyStock` method.
+
+Below is a table of what each parameter corresponds to in the state diagram of the program.
+
+|Parameter|Corresponds to
+|:---:|:---:
+|`symbol`| Ticker symbol of Stock to buy
+|`quantity`| Integer number of shares to be 
+|`price`| Price of stock at current time 
+
+**Step 2** : `buyStock()` is called from the `PortfolioManager` with the values of `symbol` and `quantity` 
+and `price` passed to them. 
+
+**Step 3** : `Portfolio` is instantiated and its `buyStock` method is called as well. 
+ 
+ ![](./diagrams/BuyStockState1.png)
+
+ 
+**Step 4** : `Wallet` and `Stock` are instantiated. 
+It then checks if there is sufficient fund in the wallet. If so, `buyStock` method is called from the `wallet` instance.
+Then, new `Transaction` object is instantiated. 
+The `transaction` object stores details of the stock bought.
+Below is a table of what each attribute in `Transaction` corresponds to in the program.
+
+|Attribute|Corresponds to
+|:---:|:---:
+|`TransactionType`| Buy or Sell stock
+|`Quantity`| Integer number of shares to be bought
+|`BuyPrice`| Cost price of a stock at a specific time 
+|`LocalDateTime`| the time when the command is called 
+
+**Step 5** : Following that, _if a stock with the same symbol has not been instantiated before_, 
+a new 'Stock' object is also instantiated. Otherwise, the `Stock` object of that stock symbol will be used. 
+
+The method `addTransaction` in the `stock` object is then called, with the `transaction` object as a parameter, 
+to update the value of the attribute `totalQuantity` in `Stock`.
+
+![](./diagrams/BuyStockState2.png)
+
+**Step 6** : `Portfolio`, `Wallet`, `Transaction` and `Stock` are terminated first.
+The `save` method is then called. 
+
+**Step 7**: `Controller` then calls the relevant methods from `Ui` to print the information about the stock bought and 
+the amount left in the wallet. `StockPriceFetcher`, `Ui` and `PortfolioManager` are then terminated.
+
+![](./diagrams/BuyStockState3.png)
+
+The following sequence diagram summarizes what happens when the user executes an `BuyCommand` :
+
+![](./diagrams/BuyStockSequence.png)
```

Comment:
Should the Portfolio, wallet, transaction and stock be destroyed when they are terminated (maybe could add a cross in the diagram) ?